04_numeronym/davidbroughsmyth/main.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"unicode/utf8"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	for i, val := range vals {
+		c := utf8.RuneCountInString(val)
+		if c > 3 {
+			f, _ := utf8.DecodeRuneInString(val)
+			l, _ := utf8.DecodeLastRuneInString(val)
+			val = fmt.Sprintf("%c%d%c", f, c-2, l)
+		}
+		result[i] = val
+	}
+	return result
```

Comment:
IMO it's nicer to factor this assignment into two functions: `numeronyms` and `numeronym`:
```go
func numeronyms(vals ...string) []string {
	result := make([]string, len(vals))
	for i, val := range vals {
                result[i] =  numeronym(val)
        }
        return result
}

func numeronyms(s string) string {
    // ...
}
```

Reply:
I did consider this but given how simple the function is, all the workings of this single function I can keep in my head. To me, this is cleaner and has greater readability.